### PR TITLE
Added cmd_vel topic for hardware abstraction. Added msg for thrust topic

### LINKED ISSUE #45: Implement `cmd_vel` topic to control thrusters

### PR DESCRIPTION
**Pull request by AVSurfer123**
- Converted the old `MotorOutput.py` to just publish to the `cmd_vel` topic. 
- Added a new `velocity_transformer` node to convert Twist messages into thrusts for each motor, assuming the Twist is a relative value. 
- Added a `pixhawk_interface` to emulate the old `MotorOutput` by reading form `cmd_vel` and publishing to Mavros OverrideRCIN

Code is still messy but will be refactored once we split into a new repository. 

Closes #45

### CODE REVIEW

File: controls/controls/MotorOutput.py [hunk 1]
```diff
@@ -4,58 +4,27 @@
...
     '''
     def __init__(self):
-
-        # publishing objects
-        self.chatter_pub = rospy.Publisher("/mavros/rc/override", OverrideRCIn, queue_size=10)
-        #self.chatter_pub = rospy.Publisher("/mavros/manual_control/send", ManualControl, queue_size=1)
-        # rate of publishing
-        #self.chat_frequency = rospy.Rate(0.25)
-        
+        self.pub = rospy.Publisher("/sub/cmd_vel", Twist, queue_size=10)
```
Comment (caelinsutch) [inline controls/controls/MotorOutput.py:15]: Is the `/sub/__` the naming style now?

File: controls/controls/MotorOutput.py [hunk 2]
```diff
@@ -67,11 +36,3 @@ def stop(self):
 
 setMotor = SetMotors()
 print("SetMotor node running")
```
Comment (caelinsutch) [inline controls/controls/MotorOutput.py:38]: Do we want to keep print?

File: controls/controls/pixhawk_interface.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+class PixhawkInterface():
+    ''' Generates and publishes ROS messages
+    '''
+
+    def __init__(self):
+        self.sub = rospy.Subscriber("/sub/cmd_vel", Twist, self.thrust_callback)
+        self.pub = rospy.Publisher("/mavros/rc/override", OverrideRCIn, queue_size=10)
+
+        #self.chatter_pub = rospy.Publisher("/mavros/manual_control/send", ManualControl, queue_size=1)
```
Comment (caelinsutch) [inline controls/controls/pixhawk_interface.py:25]: Remove comments

File: controls/controls/velocity_transformer.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        """
+
+        msg = SubThrusts()
+        msg.header.stamp = rospy.Time.now()
+        msg.thrusts[0] = twist.linear.x + twist.linear.y - twist.angular.z
+        msg.thrusts[1] = -twist.linear.x + twist.linear.y + twist.angular.z
+        msg.thrusts[2] = -twist.linear.x + twist.linear.y - twist.angular.z
+        msg.thrusts[3] = twist.linear.x + twist.linear.y + twist.angular.z
+
```
Comment (caelinsutch) [inline controls/controls/velocity_transformer.py:47]: Did you test this math?

**Comment by caelinsutch** [comment]
Plus see codefactor before merging back in

File: controls/controls/MotorOutput.py (hunk 1 above)
Comment (AVSurfer123) [reply controls/controls/MotorOutput.py:15]: Yup

File: controls/controls/MotorOutput.py (hunk 2 above)
Comment (AVSurfer123) [reply controls/controls/MotorOutput.py:38]: I'll remove it

File: controls/controls/velocity_transformer.py (hunk 4 above)
Comment (AVSurfer123) [reply controls/controls/velocity_transformer.py:47]: Will test it in practice this week when I hook it up to the Plankton simulator.

**Review by caelinsutch — APPROVED** [review]

**Pull request merged**